src/Controllers/DatabaseController.js

```diff
@@ -69,48 +69,73 @@ const isSpecialQueryKey = key => {
   return specialQuerykeys.indexOf(key) >= 0;
 };
 
-const validateQuery = (query: any): void => {
+const validateQuery = (
+  query: any,
+  skipMongoDBServer13732Workaround: boolean
```

Comment:
To make the code simpler, you can pass the adapter in here and also check engine and version (numerically) in here.